src/nvim/log.c

```diff
@@ -69,6 +70,19 @@ static bool log_path_init(void)
       || log_file_path[0] == '\0'
       || os_isdir((char_u *)log_file_path)
       || !log_try_create(log_file_path)) {
+    // Make kXDGCacheHome if it does not exist.
+    char *cachehome = get_xdg_home(kXDGCacheHome);
+    char *failed_dir = NULL;
+    if (!os_isdir((char_u *)cachehome)) {
+        int ret;
+        if ((ret = os_mkdir_recurse(cachehome, 0700, &failed_dir)) != 0) {
+          EMSG3(_(SERR "Failed to create directory %s "
```

Comment:
`SERR` (and E886) are specific to the shada code.  It looks like every "Failed to create directory" scenario currently has its own error code, so I guess we need a new one for this.

It needs to be in the 5xxx range since it's Neovim specific.  It looks like we already have some different groups of error numbers in the 5000+ range, but I don't know what the reasoning behind them is.  The error code should also be a tag in the help file at the same location as `:help NVIM_LOG_FILE`.

I think `:help NVIM_LOG_FILE` should specifically mention `$XDG_CACHE_HOME` so it's clear we honor that, rather than coincidentally choosing the same path.

Reply:
Updated the error code and docs. The test that is failing now is

```
[  ERROR   ] test/functional/options/defaults_spec.lua @ 236: startup defaults $NVIM_LOG_FILE defaults to .nvimlog if stdpath("cache") is invalid
```

Which makes sense, since now we are explicitly making stdpath cache, so I guess I need to figure out how to make an unwriteable stdpath cache in the test?

Reply:
Also will change the error code based on feedback as to what it should be, I just picked a free number.

Reply:
@bfredl has revealed the reasoning behind error codes is RNG